alpha/lib/model/assetPeer.php

```diff
@@ -851,4 +851,37 @@ public static function retrieveAudioFlavorsByEntryIdAndDimensions($entryId, arra
 		$c->add(assetPeer::TYPE, assetType::FLAVOR);
 		return assetPeer::doSelect($c);
 	}
+	/**
+	 *
+	 * @param string $entryId
+	 * @param string $tag
+	 * @return array<assets>
+	 */
+	public static function retrieveByEntryIdAndTag($entryId, $tag)
```

Comment:
should be more specific - retrievethumbAssetsByEntryIdAndTag
also make sure it returns only ready assets